hashring/hashring.go

```diff
@@ -187,71 +147,90 @@ func (r *HashRing) AddRemoveServers(add []string, remove []string) bool {
 	return changed
 }
 
-// hasServer returns true if the server exists in the ring, false otherwise
-func (r *HashRing) HasServer(address string) bool {
-	r.servers.RLock()
-	server := r.servers.byAddress[address]
-	r.servers.RUnlock()
+// addServer adds all replicas of a server to the HashRing.
```

Comment:
I think it would be desirable to have a comment indicating that the lock must be held in front of every method which has that precondition.